coralproject/talk
# Ignore User

## What does this PR do?

* Implements story [IGNORE ANOTHER USER](https://www.pivotaltracker.com/n/projects/1863625/stories/141803099)
  * There is now a little menu in the top right of Comments authored by users other than the logged in user. Click it to start the Ignore User flow.
  * Asset->comments, Comment->replies, and several count edges in the graph can now be personalized to take into account ignored users. @jde and I discussed a bit how to do this before I went forth, but we both acknowledged its a bit tricky. So if any reviewer has good ideas for altering my first draft here, I welcome the feedback.

* Implements story [MANAGE IGNORED USERS](https://www.pivotaltracker.com/n/projects/1863625/stories/141803209)
  * Rename "Settings" tab to "My profile"
  * Adds "Ignored Users" section if you are ignoring users, which allows you to stop ignoring users

![gif of this in action](https://media.giphy.com/media/l4FGqsOlvfrBlYrLy/giphy.gif)

## How do I test this PR?

Ignoring another user

1. Load the embed
2. Click on the little chevron to open the new menu in the top right of Comments *that aren't authored by you*
3. Follow the mini wizard to ignore that user
4. Once you ignore the user, all their comments are replace with a tombstone until the data for the stream is next fetched (like refresh, or switching to another tab and then back)
5. After the stream is next fetched (e.g. refresh), you no longer see the ignored users comments, nor do you see any tombstones

Managing ignored users

1. Click the 'My profile' tab to manage ignored users
2. If you aren't ignoring any users, there is no (empty) "Ignored Users" section here, don't worry about it
3. If you *are* ignoring users, there is an "Ignored Users" section listing all users you've ignored.
4. Click "Stop Ignoring" on one
5. Switch back to the comment stream tab
6. You can see that users comments again
